presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveBasicTableStatistics.java

```diff
@@ -501,22 +475,9 @@ private static BasicStatistics basicStatisticsFromDescribeResult(QueryResult res
         return new BasicStatistics(numFiles, numRows, rawDataSize, totalSize);
     }
 
-    private static Optional<String> getTableLocation(String tableName)
+    private static String getTableLocation(String tableName)
     {
-        QueryResult describeResult = onHive().executeQuery(format("DESCRIBE FORMATTED %s", tableName));
-        verify(describeResult.getColumnsCount() == 3, "describe result is expected to have 3 columns");
-        for (List<?> row : describeResult.rows()) {
-            Optional<String> name = Optional.ofNullable(row.get(0))
-                    .map(Object::toString)
-                    .map(String::trim);
-
-            if (name.isPresent() && "Location:".equals(name.get())) {
-                return Optional.ofNullable(row.get(1))
-                        .map(Object::toString)
-                        .map(String::trim);
-            }
-        }
-        return Optional.empty();
+        return onPresto().executeQuery("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName).<String>column(1).get(0);
```

Comment:
```suggestion
        return getOnlyElement(onPresto().executeQuery("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName).column(1));
```

-- fail if there are 2+ results